packages/core/src/domain/internalMonitoring/internalMonitoring.ts

```diff
@@ -35,10 +44,16 @@ let onInternalMonitoringMessageCollected: ((message: MonitoringMessage) => void)
 
 export function startInternalMonitoring(configuration: Configuration): InternalMonitoring {
   let externalContextProvider: () => Context
+  let telemetryContextProvider: () => Context
   const monitoringMessageObservable = new Observable<MonitoringMessage>()
+  const telemetryEventObservable = new Observable<TelemetryEvent & Context>()
 
-  onInternalMonitoringMessageCollected = (message: MonitoringMessage) =>
+  onInternalMonitoringMessageCollected = (message: MonitoringMessage) => {
```

Comment:
Lgtm for now. But I think that when we gonna clean old internal monitoring will be able to make `telemetryEventObservable` a singleton and use it directly instead going through `onInternalMonitoringMessageCollected`.

Reply:
Sure. All good ;)